plugins/BEdita/Core/src/Migration/ResourcesMigration.php

```diff
@@ -81,4 +107,133 @@ protected function getConnection(): Connection
     {
         return $this->getAdapter()->getCakeConnection();
     }
+
+    /**
+     * Retrieve Column Types
+     *
+     * @return array
+     *
+     * @codeCoverageIgnore
+     */
+    protected function columnTypes(): array
+    {
+        return $this->getAdapter()->getColumnTypes();
+    }
+
+    /**
+     * Extracts column related actions from migration data.
+     * Removes column actions from migration data.
+     *
+     * @param array $data Migration data.
+     * @return array
+     */
+    protected function tableColumnsActions(array &$data): array
+    {
+        $res = [];
+        foreach ($data as $action => &$value) {
+            $path = 'properties.{n}[is_column=true]';
+            $res[$action] = Hash::extract($value, $path);
```

Comment:
are actions all different? then fine.
otherwise this statement override a possible previous action in the $data array.

Reply:
yes, `$action` can only be `create`, `remove` or `update` - only once in array